terraform/modules/happy-service-eks/main.tf

```
@@ -173,6 +173,51 @@ resource "kubernetes_deployment_v1" "deployment" {
           }
         }
 
+        dynamic "container" {
+          for_each = coalesce(var.sidecars, {})
```

```suggestion
          for_each = var.sidecars
```

The sidecars variable has a default of `{}`, so it doesn't need the coalesce